beetsplug/embedart.py

```diff
@@ -107,13 +110,38 @@ def embed_func(lib, opts, args):
                     art.embed_item(self._log, item, imagepath, maxwidth,
                                    None, compare_threshold, ifempty,
                                    quality=quality)
+            elif opts.url:
+                from PIL import Image
+                try:
+                    response = requests.get(opts.url, timeout=5)
+                    response.raise_for_status()
+                except requests.exceptions.RequestException as e:
+                    self._log.error("Error: {}".format(e))
+                try:
+                    img = Image.open(BytesIO(response.content))
+                except OSError as e:
+                    self._log.error("Error: {}".format(e))
+                if img.format:
+                    with open('temp.jpg', 'wb') as f:
+                        f.write(response.content)
+                    opts.file = 'temp.jpg'
```

Comment:
I'm still not sure if it hurts to handle errors here. What happens if the file can't be written. Disk full, permission problem or something like that. Maybe I'm overcautious but  yeah...

Reply:
Good point....added more error handling.